src/synapse-plugins/calculator-plugin/calculator-plugin-backend.vala

```diff
@@ -115,5 +101,42 @@ namespace Synapse {
                 return double.parse (solution);
             }
         }
+
+
+        private bool allowed_expression (string expr, out bool is_base_expr) {
+            is_base_expr = false;
+            if (expr.length > 6 &&
+                (expr.has_prefix ("ibase=") || expr.has_prefix ("obase="))
+            ) {
+                var suffix = expr.slice (6, expr.length);
+                var valid = int.parse (suffix) > 1 && int.parse (suffix) <= 16;
+                if (valid && expr.has_prefix ("ibase=")) {
+                    use_num = VALID_NUM.slice (0, int.parse (suffix));
+                }
```

Comment:
NIT: This code repeats `int.parse` 3 times, might be good to do it once and save it in a variable?